File: app/src/main/java/com/ciastek/todoapp/TaskPriority.java
package com.ciastek.todoapp;

/**
 * Created by emcia on 19.08.2017.
 */

public enum TaskPriority {
    MINOR("Minor"),
    MAJOR("Major"),
    CRITICAL("Critical");


    private String priorityName;
    private int value;

    TaskPriority(String priorityName){
        this.priorityName = priorityName;
    }

    @Override public String toString(){
        return priorityName;
    }

    public int getValue(){
        switch (priorityName){
            case "Minor":
                return 0;
            case "Major":
                return 1;
            case "Critical":
                return 2;
            default:
                return 0;
        }
    }
}

File: app/src/main/java/com/ciastek/todoapp/model/TaskPriority.kt
package com.ciastek.todoapp.model

import android.support.annotation.IntDef
import com.ciastek.todoapp.model.PRIORITY.CRITICAL
import com.ciastek.todoapp.model.PRIORITY.MAJOR
import com.ciastek.todoapp.model.PRIORITY.MINOR

@Retention(AnnotationRetention.SOURCE)
@IntDef(MINOR, MAJOR, CRITICAL)
annotation class TaskPriority

